scripts/buildorder.py

```diff
@@ -305,32 +309,6 @@ def generate_full_buildorder(pkgs_map):
             if not remaining_deps[other_pkg.name]:  # all deps were already appended?
                 pkg_queue.append(other_pkg)  # should be processed
 
-    if set(pkgs_map.values()) != set(build_order):
-        print("ERROR: Cycle exists. Remaining: ", file=sys.stderr)
-        for name, pkg in pkgs_map.items():
-            if pkg not in build_order:
-                print(name, remaining_deps[name], file=sys.stderr)
-
-        # Print cycles so we have some idea where to start fixing this.
```

Comment:
Please do not get rid of the cycle printer, it's made debugging these issues so much easier.

Reply:
I agree with this comment, proper fix would be to update build recipes to fix cycles, so we still want this info to simplify debugging and fixes

Reply:
> Why take out the diagnostic tool if the problem may reoccur?

It diagnosed cyclic dependencies with the return of an error, which is not allowed here. And again, after changing the logic for creating the `leaf_pkgs` list, it will not launch in any way.

> I agree with this comment, proper fix would be to update build recipes to fix cycles, so we still want this info to simplify debugging and fixes

I can create a separate function in `buildorder.py` that will allow to find cyclic dependencies.

Reply:
> I can create a separate function in `buildorder.py` that will allow to find cyclic dependencies.

Then please do.

Reply:
Done